### DISCUSSION TITLE
Netcdf file extraction

### DISCUSSION THREAD

**Discussion by awel-H** [jddingemanse/awtiCodeDev#23] [Q&A] [answered]
If we have satellite data e.g. TRMM file in .NC form how do we extract this file to different stations at once using the station longitude and latitude to compare with the point station data and also for different periods of years?

**Comment by jddingemanse** [comment, marked as answer]
Nice question! General approach:
- Load a netcdf file as xarray DataSet;
- Select only the longitude and latitude closest / best representing your station location;
- After that, use methods like taking the mean over time, or resample, to change your time step.

**awtiCode package**
I have written code for selecting data from a netcdf file based on station coordinates, and included it in the package awtiCode. I would like it very much if you try out this code, and let me know whether it works or not, and whether it is clear to use or not. To be able to use it, you first need to install the package awtiCode. You can follow the instructions of [this page](https://github.com/jddingemanse/awtiCode).

After installing, what you need to use, is the function ncDataSelect from the module mixedFileTypes under awtiCode. In other words, **after installation was successful**, in Jupyter Notebook or Spyder run:
`from awtiCode import mixedFileTypes as mFT`
Then, you can access the help-text on the function `ncDataSelect` by running
`help(mFT.ncDataSelect)`
You can use the function `ncDataSelect` by supplying an Xarray DataSet, as well as coordinates. You can also select from three methods on how to select the data for those coordinates: get the closest, get the average of surrounding coordinates, or get the inverse distance weighted average of surrounding coordinates. It should give you the data for the time dimension for the selected coordinates.

If you want, you can input that data, together with your station data, into the function `alignStationNC`, to get a combined file in which both stationdata and the netcdf data are in the same DataFrame. To read the helptext on that function, run:
`help(mFT.alignStationNC`.

Let me know if you run into any problems, or if installation and using those functions works for you!
